lemarche/templates/includes/_tracker_itou.html

```diff
@@ -0,0 +1,173 @@
+    meta['user_type'] = getCookieValueByName(USER_TYPE_COOKIE_NAME);
+    meta['user_cookie_type'] = getCookieValueByName(USER_TYPE_MODAL_COOKIE_NAME);
+
+    return {
+        _v: VERSION,
+        timestamp: timeNow.toISOString(),
+        order: ORDER += 1,
+        env: '{{ BITOUBI_ENV }}',
+        session_id: getSessionId(),
+        page: page,
+        action: action,
+        meta: JSON.stringify(meta),
+        client_context: {},
+        server_context: {},
+    }
+}
+
+
+function uuidv4() {
+try {
+    return ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, c =>
+        (c ^ ( crypto.getRandomValues(new Uint8Array(1))[0] & (15 >> (c / 4)))).toString(16)
+    );
+} catch {
+    return 'ANONYMOUS';
+}
+}
+
+// export function getSessionId() {
+function getSessionId() {
+    if (SESSION_ID === "") {
+        if (typeof document !== `undefined`) {
+            SESSION_ID = (new URL(document.location)).searchParams.get('sid') || false;
+        }
+
+        if (!SESSION_ID && window.sessionId !== undefined) {
+            SESSION_ID = window.sessionId;
+        } else {
+            SESSION_ID = uuidv4();
+        }
+    }
+    return SESSION_ID;
+}
+
+function getUrlMeta() {
+    // Avoid using JSON and interpreting user-provided values
+    var meta = {};
+    const doc = typeof document !== `undefined` ? document : null;
+    try {
+        const check = (new URL(doc.location)).searchParams.get('t');
+        const split1 = check.split(';');
+        for (var tuple of split1) {
+            const keyval = tuple.split(':');
+            const key = keyval[0].replace(/[^\-_0-9a-zA-Z]/g,'');
+            const val = keyval[1].replace(/[^\-_0-9a-zA-Z]/g,'');
+            meta[key] = val;
+        }
+    } finally {
+        return meta;
+    }
+}
+
+document.addEventListener('DOMContentLoaded', () => {
+    track(window.location.pathname, Steps.LOAD);
+}, false);
+
+document.addEventListener('click', async (event) => {
+    event = event || window.event;
+    var target = event.target || event.srcElement;
+
+    var loops = 3;
+    var meta = {}
+    while (target && loops > 0) {
+        if (target instanceof HTMLAnchorElement || target instanceof HTMLButtonElement) {
+            meta = {
+                'href': target.getAttribute('href'),
+                'id' : target.id,
+                'class': target.className
+            }
+        }
+        target = target.parentNode;
+        loops -= 1;
+    }
+
+    await track(
+        window.location.pathname,
+        Steps.CLICK,
+        meta
+    );
+}, false);
+
+// _header_notice_contact.html
+// show header-notice if buyer or actor
+user_cookie_type = getCookieValueByName(USER_TYPE_MODAL_COOKIE_NAME);
+if ((user_cookie_type === "buyer") || (user_cookie_type === "actor")) {
+    const headerNotice = document.getElementById("header-notice");
+    if (headerNotice) {
+        headerNotice.removeAttribute('class');
```

Comment:
Un peu violent non ? 😅 
```suggestion
        headerNotice.classList.remove('d-none');
```